<launch>

  <arg name="robot_name" default=""/>
  <arg name="robot_namespace" default="enemy_bot"/>

  <node name="obstacle_extractor" pkg="obstacle_detector" type="obstacle_extractor_node">
    <param name="active"               value="true"/>
    <param name="use_scan"             value="true"/>
    <param name="use_pcl"              value="false"/>

    <param name="use_split_and_merge"  value="true"/>
    <param name="min_group_points"     value="3"/>
    <param name="distance_proportion"  value="0.05"/>
    <param name="max_group_distance"   value="0.05"/>
    <param name="max_split_distance"   value="0.05"/>
    <param name="max_merge_separation" value="0.2"/>
    <param name="max_merge_spread"     value="0.01"/>
    <param name="max_circle_radius"    value="0.15"/>
    <param name="radius_enlargement"   value="0.0"/>

    <param name="frame_id"             value="$(arg robot_namespace)/map"/>
  </node>

  <node name="obstacle_tracker" pkg="obstacle_detector" type="obstacle_tracker_node">
    <param name="loop_rate"         value="15"/>
    <param name="frame_id"          value="$(arg robot_namespace)/map"/>
    <remap from="tracked_obstacles" to="obstacles"/>
  </node>

</launch>
